vllm_spyre/v1/worker/spyre_model_runner.py

```diff
@@ -534,6 +535,32 @@ def _get_padded_batch_size(self, new_requests: list[NewRequestData]):
         padded_batch_size = applicable_spyre_warmup_shapes[0]['batch_size']
         return padded_batch_size, min_pad_length_batch
 
+    def _mark_input_tensors(self, model_input: ModelInputForSpyre) -> None:
+        """Yoinked from 
+        https://github.com/foundation-model-stack/aiu-fms-testing-utils/pull/13
+        """
+        # To produce like graphs during pre-fill, we mark the prefill
+        # batch x seq as static, but relax this for decode for the seq
+        if model_input.is_prompt:
+            # we always want prefill to be static to produce same-like graph
+            torch._dynamo.mark_static(model_input.input_tokens, 0)
+            torch._dynamo.mark_static(model_input.input_tokens, 1)
+            torch._dynamo.mark_static(model_input.input_masks, 0)
+            torch._dynamo.mark_static(model_input.input_masks, 1)
+            torch._dynamo.mark_static(model_input.input_masks, 2)
+            torch._dynamo.mark_static(model_input.input_positions, 0)
+            torch._dynamo.mark_static(model_input.input_positions, 1)
+        else:
+            # we always want the decode to be dynamic on sequence
```

Comment:
The FMS code has a check on `if torch._dynamo.config.dynamic_shapes:` before using `mark_dynamic`.  It looks like PyTorch will raise an exception if using `mark_dyamic` with `dynamic_shapes == False` ([REF](https://github.com/pytorch/pytorch/blob/66a76516bfc341b2b55bb2056d2faa9c2de46d69/torch/_dynamo/decorators.py#L186-L187)).

That might be fine, but wanted to point it out.

Reply:
Ah yeah, it was my understanding that we're always using dynamic shapes, and I didn't see the CB code making that check either. I can read back on some issue comments to see if that assumption is true 🤔